### PR TITLE
stabilize RBI dependency file output

### PR DESCRIPTION
**Pull request by froydnj**
### Motivation

Deterministic output is a good thing.

### Test plan

I need this for tests in #5352, so real tests for deterministic output will be added there.

### CODE REVIEW

File: packager/rbi_gen.cc [hunk 1]
```diff
@@ -1167,6 +1164,23 @@ class RBIExporter final {
         return lookupFQN(gs, fullName).asClassOrModuleRef();
     }
 
+    string buildPackageDependenciesString(const core::GlobalState &gs) {
+        vector<string> packageNames;
+        packageNames.reserve(referencedPackages.size());
+        absl::c_transform(referencedPackages, back_inserter(packageNames),
+                          [&gs](const auto &p) -> string { return p.first.show(gs); });
+        fast_sort(packageNames, [](const auto &lhs, const auto &rhs) -> bool { return lhs < rhs; });
```
Comment (jez) [inline packager/rbi_gen.cc:1172]: Are FileRef IDs stable?

Comment (jez) [reply packager/rbi_gen.cc:1172]: Do we need to sort by the file path instead?

Comment (jez) [reply packager/rbi_gen.cc:1172]: i guess it looks like this is not the first place we're sorting on these IDs, and the `operator<` definition even exists so maybe it's not worth worrying about

**Review by jez — APPROVED** [review]
sorting seems fine. i usually try to ask myself in situations like this if there's a way to make it sorted by construction, instead of calling `fast_sort` after the fact. food for thought.

File: packager/rbi_gen.cc (hunk 1 above)
Comment (froydnj) [reply packager/rbi_gen.cc:1172]: Given that we sort by `FileRef` for determinism purposes in e.g. resolver:

https://github.com/sorbet/sorbet/blob/10d09b92d34d832bef454f0436cf0b35a7003cb2/resolver/resolver.cc#L1524-L1539

(and a number of other places), I think this is OK.  It does mean that the output is not stable if you change the order files are given on the command line, but I think that is mostly OK?

**Pull request merged**